app/api/views/api_view.py

```diff
@@ -18,18 +22,71 @@ def account_info(request):
             serializer = PlayerSerializer(player)
             return success(serializer.data)
         else:
-            return unauthorized([])
+            return forbidden("")
+
+def stun_tag(request):
+    if not request.user.is_authenticated:
+        return unauthorized("")
+    
+    if request.method == "GET":
+        return notFound("")
+    
+    elif request.method == "POST":
+        player = Player.objects.get(user=request.user)
+        if not Player:
+            return forbidden("")
+        target = Player.objects.get(code=request.POST.get('code'), active = True); 
+        time = int(request.POST.get('time'))
+        if type(time) != int:
+            return badRequest("")
+        date = datetime.fromtimestamp(time)
+        print(date)
+        location = request.POST.get('location')
+        if not location:
+            location = ""
+        description = request.POST.get('description')
+        if not description:
+            description = ""
+
+        if not target:
+            return notFound("Player not found")
+        
+        tag_modifier_amount = 0
+        try:
+            tag_modifier = Modifier.objects.get(faction=player.faction, modifier_type=ModifierType.TAG)
+            tag_modifier_amount = tag_modifier.modifier_amount
+        except ObjectDoesNotExist:
+            pass
+
+        if Tag.objects.filter(initiator=player, receiver=target,
+                              tagged_at=date, location=location,
+                              description=""):
+            return conflictOrDuplicate("")
+
+        try:
+            tag = Tag.objects.create_tag(player, target, date,
+                                         location, description, tag_modifier_amount)
+            tag.active = False
+            tag.save()
+        except ValueError as err:
+            return badRequest("Error creating tag")
+
+        if player.is_human:
+            send_stun_email(request, tag)
+        else:
+            send_tag_email(request, tag)
+        return success(target.user.get_full_name())
 
 @csrf_exempt
 def view_login(request):
     if request.method == "GET":
-        return notFound([])
+        return notFound("")
     
     elif request.method == "POST":
         user = authenticate(email = request.POST.get('username'), password = request.POST.get('password'))
 
         if user is None:
-            return unauthorized([])
+            return unauthorized("")
```

Comment:
Instead of doing this, maybe set a default value of "", so that way these are way cleaner. This applies to all the helper methods!